adam-core/src/main/scala/org/bdgenomics/adam/rdd/sequence/ReadRDD.scala

```diff
@@ -0,0 +1,226 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.rdd.sequence
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models._
+import org.bdgenomics.adam.rdd.{
+  AvroGenomicRDD,
+  JavaSaveArgs
+}
+import org.bdgenomics.adam.serialization.AvroSerializer
+import org.bdgenomics.formats.avro.{
+  Read,
+  Sequence,
+  Slice,
+  Strand
+}
+import org.bdgenomics.utils.interval.array.{
+  IntervalArray,
+  IntervalArraySerializer
+}
+import org.bdgenomics.utils.misc.Logging
+import scala.reflect.ClassTag
+
+private[adam] case class ReadArray(
+    array: Array[(ReferenceRegion, Read)],
+    maxIntervalWidth: Long) extends IntervalArray[ReferenceRegion, Read] {
+
+  def duplicate(): IntervalArray[ReferenceRegion, Read] = {
+    copy()
+  }
+
+  protected def replace(arr: Array[(ReferenceRegion, Read)],
+                        maxWidth: Long): IntervalArray[ReferenceRegion, Read] = {
+    ReadArray(arr, maxWidth)
+  }
+}
+
+private[adam] class ReadArraySerializer extends IntervalArraySerializer[ReferenceRegion, Read, ReadArray] {
+
+  protected val kSerializer = new ReferenceRegionSerializer
+  protected val tSerializer = new AvroSerializer[Read]
+
+  protected def builder(arr: Array[(ReferenceRegion, Read)],
+                        maxIntervalWidth: Long): ReadArray = {
+    ReadArray(arr, maxIntervalWidth)
+  }
+}
+
+object ReadRDD {
+
+  /**
+   * Builds a ReadRDD, creating the sequence dictionary
+   * from read names and lengths.
+   *
+   * @param rdd The underlying Read RDD to build from.
+   * @return Returns a new ReadRDD.
+   */
+  def apply(rdd: RDD[Read]): ReadRDD = {
+
+    // get sequence dictionary
```

Comment:
Yeah this sucks, and is only there to support the contract of `GenomicRDD`.

We could either come up with a better fake sequence dictionary or have this not extend `GenomicRDD` (see #1173).